diff --git a/dist/server/server.js b/dist/server/server.js
index 333f66d95032cb7b2a8eba447330a475189857ad..c06e4c514558c53ef1fd0296f79f17643bf85c8f 100644
--- a/dist/server/server.js
+++ b/dist/server/server.js
@@ -108,9 +108,6 @@ async function createServer(opts) {
       });
     }
     const banner = (0, import_utils.getDevBanner)(protocol, opts.host, port);
-    console.log(banner.before);
-    import_utils.logger.ready(banner.main);
-    console.log(banner.after);
   });
   return server;
 }